src/server/server_family.h

```diff
@@ -266,6 +266,18 @@ class ServerFamily {
 
   void UpdateMemoryGlobalStats();
 
+  // Return true if no replicas are registered or if all replicas reached stable sync
+  // Used in debug populate to DCHECK insocsistent flows that violate transaction gurantees
+  bool AreAllReplicasInStableSync() const {
+    auto roles = dfly_cmd_->GetReplicasRoleInfo();
```

Comment:
I think this function takes several mutexes and better we use it not so frequently therefor I sugget to do the check in debug populate only when we start the debug populate and we can also do when we finish.
Not for every command invokation

Reply:
No objection, my rationale was that we don't care since it's not used in production. Happy to change that though.

Reply:
you are right it does not affect prod but it might slow the debug populate in our tests